### PR TITLE
Fixing the issue #598

### PR DESCRIPTION
**Pull request by websiddu**
Requested to review this commit, if you are ok with this I will port this to python also.

### CODE REVIEW

**Comment by websiddu** [comment]
I am not sure why this happens!! Can any one help me ?

```bash
    Post-build git status check...
    Ensuring no changes visible to git have been made to '' ...
    ERROR: Post-build git status check - FAILED.
    Git status reported changes to non-git-ignore'd files.
    TO REPRO: Run 'git status '.  The use git gui or git diff to see what was changed during the build.
    TO FIX: Amend your commit and rebuild. Repeat until git status reports no changes both before and after the build.
    OUTPUT FOR 'git status ':
    # HEAD detached at FETCH_HEAD
    # Changes not staged for commit:
    #   (use "git add <file>..." to update what will be committed)
    #   (use "git checkout -- <file>..." to discard changes in working directory)
    #
    # modified:   js/test/beautify-javascript-tests.js
    #
    no changes added to commit (use "git add" and/or "git commit -a")
    make: *** [status] Error 1
    The command "make" exited with 2.
    Done. Your build exited with 1. 
```

**Comment by bitwiseman** [comment]
Tests are generated during buld.  Do not edit `js/test/beautify-javascript-tests.js` directly.  Edit `test/data/javascript.js` instead.

File: js/lib/beautify.js [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
   The MIT License (MIT)
 
-  Copyright (c) 2007-2013 Einar Lielmanis and contributors.
+  Copyright (c) 2007-2015 Einar Lielmanis and contributors.
```
Comment (bitwiseman) [inline js/lib/beautify.js]: If you must do this put it in another commit and do it everywhere.  It is unrelated to this commit.

File: js/lib/beautify.js [hunk 2]
```diff
@@ -386,12 +386,13 @@
             var out = [],
```
Comment (bitwiseman) [inline js/lib/beautify.js]: Update the comment on line 380, to talk about this change.

File: test/template/node-javascript.mustache [hunk 3]
```diff
@@ -62,8 +62,8 @@ function run_javascript_tests(test_obj, Urlencoded, js_beautify, html_beautify,
         // }
 
         if (opts.indent_size === 4 && input) {
-            wrapped_input = '{\n' + input.replace(/^(.+)$/mg, '    $1') + '\n    foo = bar;\n}';
-            wrapped_expectation = '{\n' + expectation.replace(/^(.+)$/mg, '    $1') + '\n    foo = bar;\n}';
+            wrapped_input = '{\n' + input.trim().replace(/^(.+)$/mg, '    $1') + '\n    foo = bar;\n}';
```
Comment (bitwiseman) [inline test/template/node-javascript.mustache]: You really shouldn't put trim here...  It means your tests aren't testing what they intend to.

**Comment by bitwiseman** [comment]
Hmm... #598 is caused by https://github.com/beautify-web/js-beautify/blob/master/js/lib/beautify.js#L1157
If you change that `trim` to `ltrim` and this fix the tests, I think that would do it.

**Comment by websiddu** [comment]
For the above comment are you saying that the trailing spaces should be preserved ?

**Comment by bitwiseman** [comment]
Yes, the bug is a little unclear, but it is a request to *not* strip trailing spaces in comments.

**Comment by websiddu** [comment]
I don't think its a good practice to preserve the trailing spaces.. Anyways, I followed your suggestions and used `ltrim` in javadoc block which fixes the issue.. @bitwiseman

**Comment by bitwiseman** [comment]
It isn't good practice to preserve trailing spaces.  We don't except in the case of comment where it makes sense to preserve them.  
This looks good.  Thanks for seeing it through.
